engine/src/main/java/org/destinationsol/world/generators/SolSystemGenerator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.world.generators;
+
+import java.util.ArrayList;
+
+/**
+ * This class defines the contract for SolSystem generators. Any SolSystem in the game will be created from a concrete
+ * implementation of this class. Every SolSystem is given access to all the available features. Particular
+ * implementation can decide which of those features will be used
+ */
+public abstract class SolSystemGenerator {
+    //This field is protected as that allows subclasses to access it
+    protected ArrayList<FeatureGenerator> featureGenerators = new ArrayList<>();
+
+    /**
+     * This method allows {@link org.destinationsol.world.WorldBuilder} to give SolSystems access to other
+     * world features so they can be initialized and incorporated into the SolSystem
+     * @param generators The list of FeatureGenerators available
+     */
+    public void setFeatureGenerators(ArrayList<FeatureGenerator> generators) {
+        for (FeatureGenerator feature : generators) {
+            featureGenerators.add(feature);
+        }
+    }
+
+    /**
+     * This method is intended to first set up the SolSystem during world generation and then initialize all the
```

Comment:
I'm not sure I understand. Does this build the actual SolSystem? If it does, wouldn't the FeatureGenerators have been initialized beforehand?

Reply:
@IsaacLic The FeatureGenerators are the next step in the world-gen process. Meaning first the SolSystem's own features are generated (belts, position in the world, deciding which Features to use), then the FeatureGenerators themselves are built (i.e. Planets, Mazes, etc. have their own build() function, and SolSystemGenerator will finish up by calling those). I think the chart in the proposal clarifies a little bit [here](https://docs.google.com/document/d/12KnOR3guxP6OKnrY9TX_29dOwVF7fi1Paa3bJNJS4qo/edit)

Reply:
That makes sense. How about "set up the broader SolSystem" and "use FeatureGenerators to do X" ? That might clarify the relationship between the two